src/components/SidebarState.vue

```diff
@@ -34,72 +80,125 @@ export default Vue.extend({
             // nextTick is needed because app_width is 0 on created()
             let sidebarDefault = this.$state.setting('sidebarDefault');
             if (sidebarDefault && this.$state.ui.app_width > 769) {
-                this.sidebarSection = sidebarDefault;
-                this.sidebarOpen = true;
+                this.channelState.sidebarSection = sidebarDefault;
+                this.channelState.sidebarOpen = true;
+                this.queryState.sidebarOpen = true;
             }
         });
     },
     methods: {
         section() {
+            if (this.sidebarOpen === null) {
+                this.sidebarOpen = this.channelState.sidebarOpen;
+            }
             if (this.isClosed) {
                 return '';
             }
 
-            let section = this.sidebarSection;
-            let isChannel = this.$state.getActiveBuffer().isChannel();
-
-            if (section === 'settings' && isChannel) {
-                return 'settings';
-            } else if (section === 'user' && this.sidebarUser && isChannel) {
-                return 'user';
-            } else if (section === 'nicklist' && isChannel) {
-                return 'nicklist';
-            } else if (section === 'about') {
-                return 'about';
+            let buffer = this.$state.getActiveBuffer();
+            if (buffer.isChannel()) {
+                if (
+                    this.sidebarOpen &&
+                    this.sidebarSection === 'user' &&
+                    !buffer.hasNick(this.sidebarUser.nick)
+                ) {
+                    // if the channel does not have the user show the nicklist
+                    this.showSidebar('nicklist');
+                }
+                return this.sidebarSection;
+            } else if (buffer.isQuery()) {
+                if (
+                    this.sidebarOpen &&
+                    (!this.sidebarUser || this.sidebarUser.nick !== buffer.name)
+                ) {
+                    let user = this.$state.getUser(buffer.getNetwork().id, buffer.name);
+                    // if the query sidebar is open and not the current user, switch users
+                    this.showSidebar('user', { user: user });
+                }
+                return this.sidebarSection;
             }
 
             return '';
         },
         close() {
             this.sidebarOpen = false;
-            this.sidebarSection = '';
             this.sidebarUser = null;
         },
-        showUser(user) {
+        showSidebar(section, opts) {
```

Comment:
`showSidebar()` has now replaced `showUser()` while all the other `show*()` methods still exist, making this inconsistent. Either the API should use `showSidebar()` or `showUser()` should be be re-implemented. Re-implementing it would be easier for devs further down the line.